### DISCUSSION TITLE
Backlog monitoring

### DISCUSSION THREAD

**Discussion by stavvy-aamiri** [Kludex/uvicorn#1979] [Q&A]
Hi!

Is there any way to tell how many requests are in the backlog at a given point in time? I have reason to suspect that my application occasionally is receiving more traffic than it can handle. In APM I'll notice that even simple endpoints that return hardcoded responses are taking multiple seconds to respond, and all the time will be spent in python code.

From what I understand, if the ASGI app receives a request and is unable to process it the request it will sit in the 
"backlog". Is there any visibility into how much of the backlog is in use?

**Comment by babbldev** [comment]
Wondering this as well!
